pkg/codegen/internal/test/testdata/simple-enum-schema/dotnet/Enums.cs

```diff
@@ -0,0 +1,76 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System;
+using System.ComponentModel;
+using Pulumi;
+
+namespace Pulumi.PlantProvider
+{
+    [EnumType]
+    public readonly struct ContainerBrightness : IEquatable<ContainerBrightness>
+    {
+        private readonly double _value;
+
+        private ContainerBrightness(double value)
+        {
+            _value = value;
+        }
+
+        public static ContainerBrightness ZeroPointOne { get; } = new ContainerBrightness(0.1);
+        public static ContainerBrightness One { get; } = new ContainerBrightness(1);
+
+        public static bool operator ==(ContainerBrightness left, ContainerBrightness right) => left.Equals(right);
+        public static bool operator !=(ContainerBrightness left, ContainerBrightness right) => !left.Equals(right);
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object? obj) => obj is ContainerBrightness other && Equals(other);
+        public bool Equals(ContainerBrightness other) => _value == other._value;
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => _value?.GetHashCode();
```

Comment:
For `double`-based enums, the `?` isn't needed here (it is for `string`).

```suggestion
        public override int GetHashCode() => _value.GetHashCode();
```